core/domain/suggestion_services.py

```diff
@@ -588,37 +588,80 @@ def get_all_suggestions_that_can_be_reviewed_by_user(user_id):
     ])
 
 
-def get_reviewable_suggestions(user_id, suggestion_type):
-    """Returns a list of suggestions of given suggestion_type which the user
-    can review.
+def get_reviewable_suggestions_for_translate_content(
+        user_id, opportunity_summary_exp_ids):
+    """Returns a list of suggestions of type translate_content
+     matching the passed opportunity ids which the user can review.
+
+    Args:
+        user_id: str. The ID of the user.
+        opportunity_summary_exp_ids: list(str) or None.
+            The list of exploration IDs for which suggestions
+            are fetched. If the list is empty, no suggestions are
+            fetched. If the value is None, all reviewable
+            suggestions are fetched. If the list consists of some
+            valid number of ids, suggestions corresponding to the
+            ids are fetched.
+
+    Returns:
+        list(Suggestion). A list of suggestions which the given
+        user is allowed to review.
+    """
+    contribution_rights = user_services.get_user_contribution_rights(
+        user_id)
+    language_codes = (
+        contribution_rights.can_review_translation_for_language_codes)
+
+    in_review_translation_suggestions = []
+    if opportunity_summary_exp_ids is None:
+        in_review_translation_suggestions = (
+            suggestion_models.GeneralSuggestionModel
+                .get_in_review_translation_suggestions(
+                    user_id, language_codes))
+    elif len(opportunity_summary_exp_ids) > 0:
+        in_review_translation_suggestions = (
+            suggestion_models.GeneralSuggestionModel
+                .get_in_review_translation_suggestions_with_exp_ids(
+                    user_id, language_codes,
+                    opportunity_summary_exp_ids))
+
+    all_suggestions = ([
+        get_suggestion_from_model(s)
+        for s in in_review_translation_suggestions
+    ])
+    return all_suggestions
+
+
+def get_reviewable_suggestions(
+        user_id, suggestion_type, opportunity_summary_exp_ids=None):
+    """Returns a list of suggestions of given suggestion_type which
+       the user can review. If suggestion_type is translate_content,
+       then either we fetch all suggestions, return an empty list
+       or return suggestions corresponding to the opportunity ids
+       depending on the value of opportunity_summary_exp_ids.
 
     Args:
         user_id: str. The ID of the user.
         suggestion_type: str. The type of the suggestion.
+        opportunity_summary_exp_ids: list(str) or None.
+            The default value is None. If not none, it contains
```

Comment:
Nit: Capitalize "none". Also, "it contains" can be ambiguous, something like `If not None, the value is a list of...` seems clearer.

Reply:
I would go further here -- drop the optional arg from this function, and in controllers/suggestion.py, call get_reviewable_suggestions_for_translate_content directly instead, in cases where you want to pass in a list of exp ids.

Reply:
Sorry, just want to clarify, I was following this suggested in the above thread: 

> I think I would suggest making a new get_reviewable_translation_suggestions(user_id, opportunity_summary_exp_ids=None) function. Have this original function call that other function for the translations case, and have your new controller call the new function. That should be cleaner.

Did I misunderstand this? Additionally, if we call directly the translate_content function, won't we then need to distinguish between different types of suggestions in the controller layer itself? That is, if (type=translate_content) call func. for translate content otherwise call func. for question suggestions. Should we expose this lower level details about translations at the controller layer?

Reply:
My suggestion above corresponds to the "have your new controller call the new function" bit.

It's a good question -- on balance, I think it's fine to differentiate. You already have that value readily available so I don't think it's an issue.

Reply:
Yep, I don't think it's an issue with value, but more around the practice, and just so that I understand and can get it right from future, how do we distinguish what's a lower level detail and at what level do we expose it? I was following the pattern what's already established in the existing code, but I am not sure if that's right. And I am a bit confused at this stage. 

Per this comment above: 
> translate_content is a somewhat lower-level implementation detail and so would prefer something higher-level like get_reviewable_translation_suggestions

We should not mention translation_suggestions in the services layer in a function name (which controller layer is calling), since it's a lower level implementation detail. But then, is it fine to expose this detail inside a function in a layer above (i.e. in controller layer above services layer)? 

IMO and what my understanding was, only suggestions should be exposed in the controller layer, and what type of suggestion it is (whether it's a translation suggestion or a question suggestion) goes in the layer below (i.e. services layer). Is this not correct?

Reply:
My interpretation of the comment by @sagangwee on the function name is that it's an issue of wording -- notice that "translation" is still part of the proposed rename. "translate content" sounds like it refers to a specific piece of content that is being translated (and it also just sounds a bit odd grammatically). So I think he is commenting more on the difference between a "translation suggestion" and a sub-part of that suggestion (but ofc, @sagangwee, please feel free to correct this if I've misunderstood it!).

The general understanding you have (in the last paragraph) is not necessarily correct -- I don't really see an issue with "knowing" in the controller layer that suggestions are of one type or the other.

Reply:
Yep, this makes sense. But, we should probably create some documentation on how do we distinguish what's a lower level detail and at what level do we expose it. Because, it can be a bit confusing to identify what's considered a lower level detail in the first place. Because, this introduces a bit of subjectivity IMO.

Reply:
In this case, it's basically the [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter) principle (which we generally try to follow, similar to other general software engineering practices like DRY/YAGNI).

Reply:
This is fixed, sorry, I accidentally captialised the entire "none" keyword. I will fix it before merging.